File: gear-ecommerce-back/src/main/java/tht/closure/operator/model/entity/VnGeographyWard.java
package tht.closure.operator.model.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity(name = "t_vn_geography_ward")
@Getter
@Setter
public class VnGeographyWard extends AbstractEntity {

    private String name;

    private String prefix;

    @ManyToOne
    @JoinColumn
    private VnGeographyDistrict district;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn
    private VnGeographyProvince province;
}

File: gear-ecommerce-front/src/material/model/VnGeographyWard.ts
import {AbstractModel} from "./main/AbstractModel";
import {SubCatalog} from "./SubCatalog";
import {VnGeographyDistrict} from "./VnGeographyDistrict";

export interface VnGeographyWard extends AbstractModel {
    name: string;
    prefix: string;
    district?: VnGeographyDistrict;
}
